_posts/en/newsletters/2025-02-07-newsletter.md

```diff
@@ -0,0 +1,533 @@
+  increases, the algorithm increases difficulty; if there are fewer
+  parents, the difficulty decreases.  Targeting an average of two
+  parents gives the fastest consensus, according to Zawy.  Unlike
+  Bitcoin's DAA, the proposed DAA doesn't need any awareness of time;
+  however, it does require peers to ignore blocks that arrive much later
+  than other blocks in the same generation.  It's not possible to
+  achieve consensus on lateness, so ultimately DAGs with more
+  proof-of-work (PoW) are preferred over those with less PoW; the
+  developer of the DAA, Bob McElrath, [analyzed][mcelrath daa-latency]
+  the problem and examined a mitigation.
+
+  Pieter Wuille [commented][wuille prior] that the proposal appears
+  similar to a [2012 idea][miller stale] by Andrew Miller; Zawy
+  [agreed][zawy prior] and McElrath [will][mcelrath prior] update his
+  paper with a citation.  Sjors Provoost [discussed][provoost dag] the
+  complexity of handling DAG chains in Bitcoin Core's current
+  architecture, but noted that it might be easier using libbitcoin and
+  efficient using [Utreexo][topic utreexo].  Zawy extensively
+  [simulated][zawy sim] the protocol and indicated that he was working
+  on additional simulations for variants of the protocol to find the
+  best mix of tradeoffs.
+
+  The last post to the discussion thread was made about a month before this
+  summary was written, but we expect Zawy and Braidpool developers are
+  continuing to analyze and implement the protocol.
+
+## Releases and release candidates
+
+_New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates._
+
+FIXME:harding to update or remove on Thursday
+
+## Notable code and documentation changes
+
+_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
+repo], and [BINANAs][binana repo]._
+
+- [Bitcoin Core #21590][] implements a safegcd-based [modular
+  inversion][modularinversion] algorithm for MuHash3072 (see Newsletter
+  [#131][news131 muhash]), based on libsecp256k1’s implementation while adding
+  support for both 32-bit and 64-bit architectures and specializing for the
+  specific modulus. Benchmark results show an approximate 100× performance
+  improvement on x86_64, reducing MuHash’s computation from 5.8 ms to 57 μs,
+  paving the way for more efficient state validation. The implementation is
+  originally based on [Bernstein and Yang’s constant-time variant of Euclid’s
+  algorithm][bernsteinandyang].
```

Comment:
@Gustavojfe FYI: I'm going to delete this final sentence mentioning Bernstein and Yang.  Our soft policy is not to mention contributor names in this section due to merges being the result of the contributions of reviewers as much (if not more) than PR authors.  Although B&Y aren't Bitcoin contributors and this is more of an academic citation, I think a full citation here would credit the significant proving and performance optimizations made to SafeGCD by Wuille and Maxwell, who are Bitcoin contributors.  Instead of the current citation or mentioning W&M's work, I'll add a link to a previous newsletter where we previously discussed all of this (and more).